Gems/Atom/Feature/Common/Code/Source/CoreLights/DiskLightFeatureProcessor.cpp

```diff
@@ -411,7 +458,41 @@ namespace AZ
             // light stores the cosine of outerConeRadians, making the equation (radius * tan(pi/2 - acosf(cosConeRadians)).
             // This simplifies to the equation below.
             float cosConeRadians = light.m_cosOuterConeAngle;
-            light.m_bulbPositionOffset = light.m_diskRadius * cosConeRadians / sqrt(1 - cosConeRadians * cosConeRadians);
+            light.m_bulbPositionOffset = light.m_diskRadius * cosConeRadians / sqrt(1.0f - cosConeRadians * cosConeRadians);
+        }
+
+        void DiskLightFeatureProcessor::UpdateBounds(LightHandle handle)
+        {
+            DiskLightData data = m_lightData.GetData<0>(handle.GetIndex());
+
+            float radius = LightCommon::GetRadiusFromInvRadiusSquared(data.m_invAttenuationRadiusSquared);
+            AZ::Vector3 position = AZ::Vector3::CreateFromFloat3(data.m_position.data());
+            AZ::Vector3 normal = AZ::Vector3::CreateFromFloat3(data.m_direction.data());
+
+            // At greater than a 68 degree cone angle, a hemisphere will have a smaller volume than a frustum.
```

Comment:
So that ~68 degrees figure is based on a frustum with a near plane of 0 (so more of a pyramid technically). I'm sure the calculation would be a bit different if the near plane was large relative to the far plane (ie, a very dim but large disk), however I think this works well enough for the common case.

You could make the argument that we should switch to hemisphere sooner than 68 degrees - while it'd increase the number of false positives, it's also somewhat cheaper to check. What I really want is a conical frustum vs aabb test :)

Reply:
Makes sense (I suspected the value came from a pyramid), maybe a comment about how the derivation is done would be useful to make the number slightly less magical